Python/libraries/recognizers-text/recognizers_text/Matcher/simple_tokenizer.py

```diff
@@ -1,23 +1,59 @@
 from .ITokenizer import Tokenizer
+from .Token import Token
 
 
 class SimpleTokenizer(Tokenizer):
 
     def tokenize(self, _input: str) -> []:
-        pass
+        tokens = []
 
-    @staticmethod
-    def is_chinese(c: chr):
-        pass
+        if not _input:
+            return tokens
 
-    @staticmethod
-    def is_japanese(c: chr):
-        pass
+        in_token = False
+        token_start = 0
+        chars = list(_input)
+        for i in range(0, len(chars)):
 
-    @staticmethod
-    def is_korean(c: chr):
-        pass
+            c = chars[i]
+            if str.isspace(c):
+                if in_token:
+                    tokens.append(Token(token_start, i - token_start, _input[token_start: i - token_start]))
+                    in_token = False
+            elif not (str.isdigit(c) or str.isalpha(c)) or self.is_cjk(c):
 
-    @staticmethod
-    def is_cjk(c: chr):
-        pass
+                # Non-splittable currency units (as "$") are treated as regular letters. For instance, 'us$' should be
+                # a single token
+                if in_token:
+                    tokens.append(Token(token_start, i - token_start, _input[token_start: i - token_start]))
+                    in_token = False
+
+                tokens.append(Token(i, 1, _input[i:1]))
+
+            else:
+
+                if not in_token:
+                    token_start = i
+                    in_token = True
+
+        if in_token:
+            tokens.append(Token(token_start, len(chars) - token_start, _input[token_start: len(chars) - token_start]))
+
+        return tokens
+
+    def is_chinese(self, c: chr):
+        uc = c
+
+        return (uc >= 0x4E00 & uc <= 0x9FBF) or (uc >= 0x3400 & uc <= 0x4DBF)
+
+    def is_japanese(self, c: chr):
+        uc = c
```

Comment:
in the C# implementation this line is used to cast the var that we get as an unsigned short, check how you do this on python and if it is necessary